src/Jarvis.Addin.StackExchange/StackOverflow/StackOverflowProvider.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using Jarvis.Addin.StackExchange.Common;
+using Jarvis.Addin.StackExchange.Common.QueryLanguage;
+using Jarvis.Core;
+
+namespace Jarvis.Addin.StackExchange.StackOverflow
+{
+    public class StackOverflowProvider : StackExchangeProvider<StackOverflowResult>
+    {
+        private readonly Lazy<ImageSource> _icon;
+        public override string Command => "so";
+
+        public StackOverflowProvider(IStackExchangeClient stackExchangeClient, IQueryParser<SearchQuery> queryParser, IQuestionDescriptionFactory descriptionFactory)
+            : base(stackExchangeClient, queryParser, descriptionFactory)
+        {
+            _icon = new Lazy<ImageSource>(() =>
+                    new BitmapImage(new Uri("pack://application:,,,/Jarvis.Addin.StackExchange;component/Resources/StackOverflow.png"))
+            );
+        }
+
+        protected override Task<ImageSource> GetIconAsync(StackOverflowResult result)
+        {
+            return Task.FromResult(_icon.Value);
```

Comment:
Instead of explicitly adding the icons to the assembly, perhaps we can download the [favicon](https://cdn.sstatic.net/Sites/stackoverflow/img/apple-touch-icon.png?v=c78bd457575a) and cache it locally? That would make it easier to add new stack exchange sites and to make the provider more dynamic.